src/baseband.c

```diff
@@ -85,7 +85,7 @@ float magnitude_true_cu8(uint8_t const *iq_buf, uint16_t *y_buf, uint32_t len)
     for (i = 0; i < len; i++) {
         int16_t x = iq_buf[2 * i] - 128;
         int16_t y = iq_buf[2 * i + 1] - 128;
-        y_buf[i]  = (uint16_t)(sqrt(x * x + y * y) * 128.0); // max 181, scaled 23170, fs 16384
+        y_buf[i]  = (uint16_t)(sqrt((double)x * x + (double)y * y) * 128.0); // max 181, scaled 23170, fs 16384
```

Comment:
This should be fixed point but I guess there is a reason for the 128.0.

Reply:
I believe the reason is more about `sqrt` taking a `double` instead of a `float`

Reply:
This is just a test function, still turning int muls into double muls isn't great.

Reply:
And same here with sqrtf().